tests/sentry/middleware/test_devtoolbar.py

```diff
@@ -0,0 +1,245 @@
+from functools import cached_property
+from unittest.mock import MagicMock, patch
+
+from django.http import HttpResponse
+from django.test import RequestFactory, override_settings
+
+from sentry.api import DevToolbarApiRequestEvent
+from sentry.middleware.devtoolbar import DevToolbarAnalyticsMiddleware
+from sentry.testutils.cases import APITestCase, SnubaTestCase, TestCase
+from sentry.testutils.helpers import override_options
+from sentry.types.group import GroupSubStatus
+
+
+class DevToolbarAnalyticsMiddlewareUnitTest(TestCase):
+    middleware = cached_property(DevToolbarAnalyticsMiddleware)
+    analytics_event_name = DevToolbarApiRequestEvent.type
+
+    @cached_property
+    def factory(self):
+        return RequestFactory()
+
+    def setUp(self):
+        # Allows changing the get_response mock for each test.
+        self.get_response = MagicMock(return_value=HttpResponse(status=200))
+        self.middleware.get_response = self.get_response
+
+    def make_toolbar_request(
+        self,
+        path="/",
+        method="GET",
+        headers=None,
+        incl_toolbar_header=True,
```

Comment:
Nit: next time it could be a good idea to skip this kind of arg and just pass in the headers that are expected in each test case.  We usually want test cases to be extremely straight forward and easy to read so it's easier to verify that they're testing the right thing. Having the default value and the if-statement in the this helper body get in the way of that a little. 
Another approach could be to have a helper called `make_request()` that doesn't add `queryReferrer`, and `make_toolbar_request()` can always, inconditionally add the extra header.

big nit though! just some background to think about for the next set of tests!

Reply:
Makes sense, thanks for the tip!